### PR TITLE
feat(components/badge): add ids

### PR DESCRIPTION
**Pull request by sdwvit**
**What is the problem this PR is trying to solve?**
QA want ids for badges to complete testing

**What is the chosen solution to this problem?**
add id props for onselect and ondelete badge

**Please check if the PR fulfills these requirements**
- [ ] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [x] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

**Comment by jmfrancois** [comment]
should we prefix id; sth like tc-badge-${id} ?

File: packages/components/src/Badge/Badge.component.js [hunk 1]
```diff
@@ -54,7 +66,9 @@ Badge.propTypes = {
 	label: PropTypes.string,
 	category: PropTypes.string,
 	onDelete: PropTypes.func,
+	deleteId: PropTypes.string,
```
Comment (jsomsanith) [inline packages/components/src/Badge/Badge.component.js]: Please avoid having multiple ids as input. You can align it with other components we have : 
* get a unique id from props
* if present, generate a delete id and a select id from it, and set it

Comment (jsomsanith) [reply packages/components/src/Badge/Badge.component.js]: + add a unit test please

Comment (sdwvit) [reply packages/components/src/Badge/Badge.component.js]: @jsomsanith thanks, updating..

**Comment by sdwvit** [comment]
@jmfrancois @jsomsanith @acateland updated

**Pull request merged**